infra/provider.tf

```diff
@@ -0,0 +1,27 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = var.region
+}
+
+# provider "kubernetes" {
+#   host                   = data.aws_eks_cluster.totem_express_cluster.endpoint
+#   cluster_ca_certificate = base64decode(data.aws_eks_cluster.totem_express_cluster.certificate_authority[0].data)
+#   token                  = data.aws_eks_cluster_auth.totem_express_cluster_auth.token
+# }
+
+# data "aws_eks_cluster" "totem_express_cluster" {
+#   name = "${var.prefix}-${var.cluster_name}"
+# }
+
+# data "aws_eks_cluster_auth" "totem_express_cluster_auth" {
+#   name = data.aws_eks_cluster.totem_express_cluster.name
+# }
```

Comment:
deixei comentado, porque precisamos dar o primeiro apply (subir as config na aws) e depois descomentar e dar um init, que ele já entende bem. como precisaremos usar outra conta, por conta da grana na minha aws academy :/

Reply:
Aliás Aninha, sobre a grana da Academy: vi que vc tá usando a `t3a.medium` no código. Algum motivo especial pra usar ela ao invés da `t2.micro`? Pq a t2 tá inclusa no free tier

Reply:
Então, acabei utilizando por recomendação do professor mesmo. Segundo ele, não teriamos imprevistos utilizando essa instancia. Não tive um motivo específico. 

Acha que vale a pena utilizar o t2?

Reply:
@analuijesus acho que vale tentar sim, porque pelo menos em termos de especificação, acho que ela dá conta sozinha num esquema de 1 pod/máquina, e daí iria escalando mais máquina à medida que for necessário. O que cê acha?